File: frontend/src/lib/admin/auditSeverity.ts
/**
 * 감사 이벤트 severity UI helper — backend `audit_log.severity` (V19) 합류 후 type 과
 * label 만 보유 (graceful upgrade 완성, P3 — Audit severity backend).
 *
 * <p>이전에는 `severityOf(eventType)` 가 `AuditEventType` 을 frontend 매핑표로 분류했으나,
 * backend `AuditSeverityMapper` 가 단일 진실이 되었고 wire 응답에 severity 가 포함된다.
 * 컴포넌트는 `entry.severity` 를 직접 사용한다.
 *
 * <p>본 모듈에 남은 책임: SeverityTab 의 'all' 필터 상태 타입과 한글 라벨 매핑.
 */

import type { AuditSeverity } from '@/types/audit'

export type { AuditSeverity } from '@/types/audit'

/** SeverityTab filter 값 — UI 상태(전체 포함). */
export type SeverityFilter = 'all' | AuditSeverity

/** 한글 라벨 — SeverityTab 표시용. */
export const SEVERITY_LABEL: Record<SeverityFilter, string> = {
  all: '전체',
  danger: '긴급',
  warn: '주의',
  info: '정보',
}

File: backend/src/main/java/com/ibizdrive/audit/AuditSeverity.java
package com.ibizdrive.audit;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

import java.util.Map;
import java.util.stream.Stream;

/**
 * 감사 이벤트 심각도 (docs/03 §4, P3 — Audit severity backend).
 *
 * <p>분류 의미:
 * <ul>
 *   <li>{@link #INFO}: 일상 운영 이벤트 (업로드/다운로드/이름 변경 등). 기본값.</li>
 *   <li>{@link #WARN}: 외부 도메인 공유 / 권한 회수 / 권한 만료 / 대량 삭제·purge / 관리자
 *       정책 변경. 검토 대상이지만 즉시 위험은 아님.</li>
 *   <li>{@link #DANGER}: 외부 노출 가능성·정책 위반 가능성. 운영자가 즉시 검토해야 하는 이벤트.</li>
 * </ul>
 *
 * <p>{@code event_type → severity} 매핑은 {@link AuditSeverityMapper}가 단일 진실. wire format 은
 * {@code lower-case} ({@code info} / {@code warn} / {@code danger}) — frontend
 * {@code AuditSeverity} 유니언 타입과 1:1 동기.
 */
public enum AuditSeverity {

    INFO("info"),
    WARN("warn"),
    DANGER("danger");

    private static final Map<String, AuditSeverity> BY_WIRE =
        Stream.of(values()).collect(java.util.stream.Collectors.toMap(AuditSeverity::wire, s -> s));

    private final String wire;

    AuditSeverity(String wire) {
        this.wire = wire;
    }

    @JsonValue
    public String wire() {
        return wire;
    }

    @JsonCreator
    public static AuditSeverity from(String wire) {
        AuditSeverity s = BY_WIRE.get(wire);
        if (s == null) {
            throw new IllegalArgumentException("Unknown AuditSeverity wire format: " + wire);
        }
        return s;
    }
}
